src/smolagents/agents.py

```diff
@@ -1011,27 +1015,33 @@ def step(self, memory_step: ActionStep) -> Union[None, Any]:
         memory_step.model_input_messages = memory_messages.copy()
 
         try:
-            model_message: ChatMessage = self.model(
-                memory_messages,
-                tools_to_call_from=list(self.tools.values()),
+            chat_message: ChatMessage = self.model.generate(
```

Comment:
To maintain backward compatibility:
```suggestion
            chat_message: ChatMessage = self.model(
```